crypto/store/store_lib.c

```diff
@@ -393,6 +407,112 @@ void STORE_INFO_free(STORE_INFO *store_info)
     }
 }
 
+int STORE_supports_search(STORE_CTX *ctx, int search_type)
+{
+    STORE_SEARCH tmp_search;
+
+    if (ctx->loader->find == NULL)
+        return 0;
+    tmp_search.search_type = search_type;
+    return ctx->loader->find(NULL, &tmp_search);
+}
+
+/* Search term constructors */
+STORE_SEARCH *STORE_SEARCH_by_name(X509_NAME *name)
+{
+    STORE_SEARCH *search = OPENSSL_zalloc(sizeof(*search));
+
+    if (search == NULL) {
+        STOREerr(STORE_F_STORE_SEARCH_BY_NAME, ERR_R_MALLOC_FAILURE);
+        return NULL;
+    }
+
+    search->search_type = STORE_SEARCH_BY_NAME;
+    search->name = name;
+    return search;
+}
+
+STORE_SEARCH *STORE_SEARCH_by_issuer_serial(X509_NAME *name,
+                                            const ASN1_INTEGER *serial)
+{
+    STORE_SEARCH *search = OPENSSL_zalloc(sizeof(*search));
+
+    if (search == NULL) {
+        STOREerr(STORE_F_STORE_SEARCH_BY_ISSUER_SERIAL, ERR_R_MALLOC_FAILURE);
+        return NULL;
+    }
+
+    search->search_type = STORE_SEARCH_BY_ISSUER_SERIAL;
+    search->name = name;
+    search->serial = serial;
+    return search;
+}
+
+STORE_SEARCH *STORE_SEARCH_by_key_fingerprint(const unsigned char *bytes,
+                                              int len)
+{
+    STORE_SEARCH *search = OPENSSL_zalloc(sizeof(*search));
+
+    if (search == NULL) {
+        STOREerr(STORE_F_STORE_SEARCH_BY_KEY_FINGERPRINT, ERR_R_MALLOC_FAILURE);
+        return NULL;
+    }
+
+    search->search_type = STORE_SEARCH_BY_KEY_FINGERPRINT;
+    search->string = bytes;
+    search->stringlength = len;
+    return search;
+}
+
+STORE_SEARCH *STORE_SEARCH_by_alias(const char *alias)
+{
+    STORE_SEARCH *search = OPENSSL_zalloc(sizeof(*search));
+
+    if (search == NULL) {
+        STOREerr(STORE_F_STORE_SEARCH_BY_ALIAS, ERR_R_MALLOC_FAILURE);
+        return NULL;
+    }
+
+    search->search_type = STORE_SEARCH_BY_ALIAS;
+    search->string = (unsigned char *)alias;
+    search->stringlength = strlen(alias);
+    return search;
+}
+
+/* Search term destructor */
+void STORE_SEARCH_free(STORE_SEARCH *search)
+{
+    OPENSSL_free(search);
+}
+
+/* Search term accessors */
+int STORE_SEARCH_get_type(const STORE_SEARCH *criterium)
```

Comment:
I had to look it up!  My primary language in this case being Swedish, I thought criterium would be the same as [kriterrium](https://en.wiktionary.org/wiki/kriterium#Swedish) (which, incidently, I just discovered has the horce race as second meaning)...  I've gone with that for ages.

Learn something new every day ;-)

I'll fix it up